### DISCUSSION TITLE
Consider using `System.Text.Json` instead of `Newtonsoft.Json` in the Arcus repo's

### DISCUSSION THREAD

**Discussion by stijnmoreels** [arcus-azure/arcus#171] [Ideas]
We use the JSON.NET package in several places, while we should consider maybe using the System.Text.Json instead.

**Comment by fgheysels** [comment]
How would this affect the boilerplate code ?

I do have projects where I've switched from System.Text.Json back to Newtonsoft because of some issues I ran into.
However, this should not matter for our templates; if we use System.Text.Json and ppl want to use Newtonsoft.Json, they're offcourse free to change it.

**Reply by stijnmoreels** [reply]
In our templates, we can definitely choose to use System.Text.Json. In the other projects, I'm not sure. This should/is not be a breaking change, if implemented correctly. I also think that we abstracted away any JSON serialization options so we're not tightly-coupled to any package. So, it seems like a good idea to move forward with this.

**Reply by fgheysels** [reply]
I agree.

**Reply by stijnmoreels** [reply]
There are, though, some functionalities that are not (yet) supported.
https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to?pivots=dotnet-6-0

**Reply by stijnmoreels** [reply]
Also, for messaging it would best if we wait when there's some kind of error handling available: https://github.com/dotnet/runtime/issues/38049
